cmake/Modules/FindLibClang.cmake

```diff
@@ -53,14 +54,14 @@ if(NOT LIBCLANG_LDFLAGS)
     # -rpath is relative path for all linked libraries.
     # The second "." is argument to rpath.
 
-    if(UNIX)
-        set(LIBCLANG_LDFLAGS_OS "-L--enable-new-dtags -L-rpath=.")
-    elseif(APPLE)
-        set(LIBCLANG_LDFLAGS_OS "-L-rpath=${LIBCLANG_DIR_PATH}")
+    if(UNIX AND NOT APPLE)
```

Comment:
Yup, UNIX is always true on Mac, to differentiate them one would need to check APPLE too